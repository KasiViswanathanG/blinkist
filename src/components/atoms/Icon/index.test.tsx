import {screen, fireEvent, getByText, render } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'

import IconComponent from '.'
import SearchIcon from '@mui/icons-material/Search';
import RocketIcon from '@mui/icons-material/Rocket';
import EntrepreneurshipPage from '../../../pages/Entrepreneurship';

describe('Testing the IconComponent', () => {
  it('render default IconComponent', () => {
    const icon = renderer.create(<IconComponent icon={undefined} title={''} titleColor={''} onClick={() => {}} />).toJSON();
    expect(icon).toMatchSnapshot();
  });

  it('render IconComponent', () => {
    const icon = renderer.create(<IconComponent icon={<SearchIcon/>} title={''} titleColor={''} onClick={() => {}} />).toJSON();
    expect(icon).toMatchSnapshot();
  });

  it('render IconComponent with title', () => {
    const icon = renderer.create(<IconComponent icon={<RocketIcon />} title={'Entrepreneurship'} titleColor={''} onClick={() => {}} />).toJSON();
    expect(icon).toMatchSnapshot();
  });
})