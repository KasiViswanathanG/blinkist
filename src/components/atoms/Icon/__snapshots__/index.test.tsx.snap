// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Button render default button 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1kjyx0o-MuiTypography-root"
  />
</div>
`;

exports[`Testing the IconComponent render IconComponent 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="SearchIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
    />
  </svg>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1kjyx0o-MuiTypography-root"
  />
</div>
`;

exports[`Testing the IconComponent render IconComponent with title 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="RocketIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2.5s4.5 2.04 4.5 10.5c0 2.49-1.04 5.57-1.6 7H9.1c-.56-1.43-1.6-4.51-1.6-7C7.5 4.54 12 2.5 12 2.5zm2 8.5c0-1.1-.9-2-2-2s-2 .9-2 2 .9 2 2 2 2-.9 2-2zm-6.31 9.52c-.48-1.23-1.52-4.17-1.67-6.87l-1.13.75c-.56.38-.89 1-.89 1.67V22l3.69-1.48zM20 22v-5.93c0-.67-.33-1.29-.89-1.66l-1.13-.75c-.15 2.69-1.2 5.64-1.67 6.87L20 22z"
    />
  </svg>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1kjyx0o-MuiTypography-root"
  >
    Entrepreneurship
  </p>
</div>
`;

exports[`Testing the IconComponent render default IconComponent 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1kjyx0o-MuiTypography-root"
  />
</div>
`;
